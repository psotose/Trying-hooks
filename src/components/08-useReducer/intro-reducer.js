const initialState = [{
  id: 1,
  todo: 'Comprar pan',
  done: false
}];

const myFirstTodoReducer = (state = initialState, action) => {

  if ( action?.type === 'add' ){
    return [ ...state, action.payload ];
  }
  return state;
}

let todos = myFirstTodoReducer();

const newTodo = [{
  id: 2,
  todo: 'Comprar leche',
  done: false
}];

const addTodoAction = {
  type: 'add',
  payload: newTodo,
}

todos = myFirstTodoReducer( todos, addTodoAction);

console.log(todos);