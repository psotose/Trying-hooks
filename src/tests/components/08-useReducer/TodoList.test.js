import React from 'react';
import { shallow } from 'enzyme';
import { demoTodos } from '../../fixtures/demoTodos';
import { TodoList } from '../../../components/08-useReducer/TodoList';

describe('Testing <TodoList />', () => {

  const handleDelete = jest.fn();
  const handleToggle = jest.fn();

  const wrapper = shallow( 
    <TodoList 
      todos={ demoTodos }
      handleDelete={ handleDelete }
      handleToggle={ handleToggle }
    /> 
  );

  test('should render correctly', () => {
    expect( wrapper ).toMatchSnapshot();
  });
  test('should have 2 <TodoListItem />', () => {
    expect( wrapper.find('TodoListItem').length ).toBe( demoTodos.length);
    expect( wrapper.find('TodoListItem').at(0).prop('handleDelete') ).toEqual( expect.any(Function) );
  })

})
