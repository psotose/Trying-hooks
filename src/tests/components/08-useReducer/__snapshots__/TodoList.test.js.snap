// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <TodoList /> should render correctly 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Learn React",
        "donde": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Learn Mongo",
        "donde": false,
        "id": 2,
      }
    }
  />
</ul>
`;
